### PR TITLE
Add frontend stack

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Richie distributes base frontend integration. We need a way to develop custom themes based on Richie's work. This task requires to set up a frontend stack to work with for fun-mooc.

## Proposal

- [x] add node-based dockerized stack to work on styles
- [x] automate frontend quality checks & testing in the CI

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,14 +44,43 @@ logs: ## display app logs (follow mode)
...
+
+build-sass: ## build Sass files to CSS
+	@$(YARN) sass
+.PHONY: build-sass
+
+install-front: ## install front-end dependencies
+	@$(YARN) install;
+.PHONY: install-front
+
+lint-front: lint-front-prettier ## run all front-end linters
```
Comment (lunika) [inline Makefile]: A preriquisite is missing or this task is useless. Just executing `lint-front` do nothing more than executing `lint-front-prettie`

File: .circleci/config.yml [hunk 2]
```diff
@@ -133,6 +133,43 @@ jobs:
...
+    steps:
+      - checkout:
+          path: ~/fun
+      - restore_cache:
+          keys:
+            - v1-front-dependencies-{{ checksum "yarn.lock" }}
+            - v1-front-dependencies-
+      - run:
+          name: Install frontend dependencies
+          command: make install
```
Comment (lunika) [inline .circleci/config.yml]: There is no `Makefile` in `src/frontend` and no task `install` in the main `Makefile`. I suppose it should be `yarn install` and your should use the option `--frozen-lockfile`

File: .circleci/config.yml [hunk 3]
```diff
@@ -133,6 +133,43 @@ jobs:
...
+      - run:
+          name: Build application styles
+          command: make build-sass
```
Comment (lunika) [inline .circleci/config.yml]: There is no `Makefile` in `src/frontend`. You should run `yarn sass` or `yarn sass-production` depending what you want.

File: .circleci/config.yml [hunk 4]
```diff
@@ -133,6 +133,43 @@ jobs:
             make migrate
             make check
 
+  # ---- Front-end jobs ----
+  build-front:
```
Comment (lunika) [inline .circleci/config.yml]: This task is not executed in the `workflow` section.

File: .circleci/config.yml [hunk 5]
```diff
@@ -133,6 +133,43 @@ jobs:
...
+          command: make install
+      - run:
+          name: Build application styles
+          command: make build-sass
+      - save_cache:
+          paths:
+            - ./node_modules
+          key: v1-front-dependencies-{{ checksum "yarn.lock" }}
+
+  lint-prettier:
```
Comment (lunika) [inline .circleci/config.yml]: This task is not executed in the `workflow` section

File: Makefile (hunk 1 above)
Comment (jmaupetit) [reply Makefile]: I'm planning for the future dude.

File: .circleci/config.yml (hunk 5 above)
Comment (jmaupetit) [reply .circleci/config.yml]: This is why it does not fail :grin:

File: Makefile (hunk 1 above)
Comment (lunika) [reply Makefile]: YAGNI

Comment (jmaupetit) [reply Makefile]: Consider this as an alias.

File: .circleci/config.yml [hunk 6]
```diff
@@ -133,6 +133,43 @@ jobs:
...
+    working_directory: ~/fun/src/frontend/
+    steps:
+      - checkout:
+          path: ~/fun
+      - restore_cache:
+          keys:
+            - v1-front-dependencies-{{ checksum "yarn.lock" }}
+            - v1-front-dependencies-
+      - run:
+          name: Install frontend dependencies (production mode)
```
Comment (lunika) [inline .circleci/config.yml]: No this is not production mode, you install `devDependencies`.

**Pull request merged**